content/articles/2018-01-fs-1.0.0.Rmd

```diff
@@ -0,0 +1,176 @@
+---
+title: fs 1.0.0
+date: '2018-01-17'
+slug: fs-1.0.0
+author: Jim Hester
+categories: [package]
+description: >
+  fs 1.0.0 is now available on CRAN. fs provides a cross-platform, uniform
+  interface to file system operations.
+photo:
+  url: https://unsplash.com/photos/8EzNkvLQosk
+  author: Maarten van den Heuvel
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  comment = "#>",
+  collapse = TRUE
+)
+knitr::opts_knit$set(
+  root.dir = system.file(package = "fs"))
+library(fs)
+```
+
+
+[fs 1.0.0][fs] is now available on CRAN!
+[fs] provides a cross-platform, uniform interface to file system operations.
+[fs] uses [libuv](http://docs.libuv.org/en/v1.x/fs.html) under the hood which
+gives a rock solid cross-platform interface to the filesystem.
+
+
+Install the latest version with:
+``` r
+install.packages("fs")
+```
+
+### Comparison vs base equivalents
+
+While it is true R comes bundled with functions for interacting with the
+filesystem, we feel **fs** improves upon them in the following ways.
+
+* All **fs** functions are vectorized, accepting multiple paths as input.
+  Base functions are inconsistently vectorized.
+
+* All **fs** functions return a character vector of paths, a named integer or a
+  logical vector (where the names give the paths). Base return values are more
+  varied and typically return error codes which need to be manually checked.
+
+* If **fs** operations fail, they throw an error. Base functions
+  tend to generate a warning and a system dependent error code. This makes it
+  easy to miss a failure.
+
+* **fs** functions always convert input paths to UTF-8 and return results as
+  UTF-8. This gives you path encoding consistency across OSs. Base functions
+  rely on the native system encoding.
+
+* **fs** functions use a consistent naming convention. Because base R's functions
+  were gradually added over time there are a number of different conventions
+  used (e.g. `path.expand()` vs `normalizePath()`; `Sys.chmod()` vs 
+  `file.access()`).
+
+
+### Tidy paths
+
+**fs** functions always return 'tidy' paths. Tidy paths always expand `~`, use
+`/` to delimit directories and never have multiple `/` or trailing `/`.
+
+Tidy paths are also coloured (if your terminal supports it) based on the
+file permissions and file type. This colouring can be customised or extended by
+setting the `LS_COLORS` environment variable, in the same output format as [GNU
+dircolors](http://www.bigsoft.co.uk/blog/index.php/2008/04/11/configuring-ls_colors).
+
+![](/images/fs-1.0.0/colour.png)
+
+### Usage
+
+**fs** functions are divided into four main categories: manipulating paths
```

Comment:
Possibly just "paths" for list agreement, since "files, directories, and links" are all nouns? (clarity over stupid grammar rules, though, of course)